src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs

```diff
@@ -36,13 +36,20 @@ public Manifest(ManifestMetadata metadata, ICollection<ManifestFile> files)
             if (files != null)
             {
                 Files = files;
+                HasFilesNode = true;
+            }
+            else
+            {
+                HasFilesNode = false;
             }
         }
 
         public ManifestMetadata Metadata { get; }
 
         public ICollection<ManifestFile> Files { get; } = new List<ManifestFile>();
 
+        public bool HasFilesNode { get; }
```

Comment:
Is `Files` being null that same indicator? Also, could `HasFilesNode` be implemented as `get { return Files == null` }`?